### PR TITLE
refactor(Core/Spell): Use ApplySpellFix with SpellInfo

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
- Replace SpellEntry to SpellInfo in ApplySpellFix
## Tests Performed:

- builds
- tested in-game, spell fixes still work as they should

## How to Test the Changes:

1. Just test that the spells that are fixed with ApplySpellFix works the same
    - Clarification, find any spell ID within ApplySpellFix and test that it works the exact same as before when SpellEntry was used 

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
Hmm ...

**Comment by Kitzunu** [comment]
Odd that we don't get the SpellInfo from any spell..

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -3422,18 +3422,18 @@ void SpellMgr::LoadSpellCustomAttr()
     LOG_INFO("server.loading", " ");
 }
 
-inline void ApplySpellFix(std::initializer_list<uint32> spellIds, void(*fix)(SpellEntry*))
+inline void ApplySpellFix(std::initializer_list<uint32> spellIds, void(*fix)(SpellInfo*))
 {
     for (uint32 spellId : spellIds)
     {
-        SpellEntry const* spellInfo = (SpellEntry*)sSpellStore.LookupEntry(spellId);
+        SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(spellId);
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellMgr.cpp]: Allways returns null. Why

**Comment by Winfidonarleyan** [comment]
Welcome more crash

**Comment by Kitzunu** [comment]
?

**Comment by Kitzunu** [comment]
Ready for review and test

**Comment by Winfidonarleyan** [comment]
Need test Nax with this PR

**Comment by Kitzunu** [comment]
What is specific to that script that I change with this PR?

**Comment by Winfidonarleyan** [comment]
> What is specific to that script that I change with this PR?

Yes

**Comment by Kitzunu** [comment]
I love your answer, but it doesn't really answer my question xD

**Comment by Kitzunu** [comment]
I have tested a couple of spell fixes and they work as they should

**Comment by claudiodfc** [comment]
you could at least have written a test case, your pr's are too hard to test because it's always very vague and confusing. Start being more specific thanks.

**Review by Nyeriah — APPROVED** [review]

**Comment by claudiodfc** [comment]
image

**Comment by Kitzunu** [comment]
> Just test that the spells that are fixed with ApplySpellFix works the same

This is not vague. This means, take any spell that you find in ApplySpellFix and test that it still has the spell fixes that are applied.

Aka the fixed spells are still fixed

**Comment by Kitzunu** [comment]
> image

Now this is vague because I cannot tell how you caused this

**Comment by claudiodfc** [comment]
1. You assume people know what ApplySpellFix is or SpellInfo is.
2. Forcing people to search the files for the affected code is just gonna end up with either no tests or bad tests.
3. This PRs changes are _"Finally..."_ You think this isn't vague?
4. I compiled the core changes and launched **worldserver**, producing this crash.
5. I'm done testing this and any of your Pull requests because you force people to follow a Pull request template that you don't follow, all you do is copy shit from Trinity Core for no reason. Have fun merging this.

**Comment by Kitzunu** [comment]
> all you do is copy shit from Trinity Core for no reason

Okay <3

**Comment by claudiodfc** [comment]
Don't bother, I won't test code that has no purpose.

**Comment by Kitzunu** [comment]
Oh but purpose it does have

**Comment by claudiodfc** [comment]
Yes, adding one more number to your name on useless pull requests and lines of code changes, congratulations.

**Comment by Kitzunu** [comment]
Ok, man I am getting sick of it now. If you do not know what the code does, which clearly you have proven not to do, then do not bother with it. Go test PRs that are easier and more understandable please.

**Review by Nyeriah — APPROVED** [review]

**Comment by Winfidonarleyan** [comment]
Tested with `Sapphiron`, no crash, good

**Pull request merged**